go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-operator
 
-go 1.21
+go 1.22
```

Comment:
Do  we want to define a guideline like on the collector when to bump this version?
See: https://github.com/open-telemetry/opentelemetry-collector#compatibility